src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsCodeFixProvider.cs

```diff
@@ -0,0 +1,257 @@
+                            }
+                        }
+                    }
+
+                    break;
+                case DiagnosticIds.PreferDeclarationInitializersExceptNullOrZero:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: "Do not initialize to null or zero",
+                            createChangedSolution: cancellationToken => EnforceFieldIsNotInitializedAsync(context.Document, declaration, cancellationToken),
+                            equivalenceKey: "Use member initializer"),
+                        diagnostic);
+                    break;
+                case DiagnosticIds.PreferDeclarationInitializersExceptWhenVaryingInitializations:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: "Do not use field initializer when varying initializations",
+                            createChangedSolution: cancellationToken => EnforceFieldIsNotInitializedAsync(context.Document, declaration, cancellationToken),
+                            equivalenceKey: "Use member initializer"),
+                        diagnostic);
+                    break;
+                case DiagnosticIds.PreferDeclarationInitializersWhenNoInitializationPresent:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: "Use initializer in field declaration",
+                            createChangedSolution: cancellationToken => EnforceFieldDeclarationInitializationAsync(context.Document, declaration, cancellationToken),
+                            equivalenceKey: "Use member initializer"),
+                        diagnostic);
+                    break;
+                default:
+                    continue;
+            }
+        }
+
+        foreach (IFieldSymbol fieldSymbol in initializationsToFixPerField.Keys)
+        {
+            if (initializationsToFixPerField.TryGetValue(fieldSymbol, out IList<ExpressionStatementSyntax> initializations))
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: "Initialize in field declaration instead",
+                        createChangedSolution: cancellationToken => ReplaceAssignmentsWithFieldDeclarationInitializerAsync(context.Document, fieldSymbol, initializations, cancellationToken),
+                        equivalenceKey: "Use member initializer"),
+                    context.Diagnostics);
+            }
+        }
+    }
+
+    private static async Task<Solution> EnforceFieldIsNotInitializedAsync(Document document, SyntaxNode declaration, CancellationToken cancellationToken)
+    {
+        SyntaxNode? root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root is not null && declaration is FieldDeclarationSyntax fieldDeclaration)
```

Comment:
nit: reduce nesting

Reply:
I have yet to make any improvements on the provider. It is still in its V1 draft. When I do that, I'll address all of the provider related comments as well

Reply:
Done. Let me know if I took it to the extreme, since I basically crammed a lot of checks into one if clause. This was to avoid duplicating the same failure case code, but I would not be surprised if there's a cleaner way